src/main/java/org/testng/TestNG.java

```diff
@@ -844,12 +847,10 @@ private void initializeCommandLineSuitesParams() {
     }
 
     for (XmlSuite s : m_cmdlineSuites) {
-      if (m_threadCount != -1) {
+      if(m_useThreadCount) {
         s.setThreadCount(m_threadCount);
```

Comment:
FYI and for future pull requests, it's better to remove this kind of syntactic changes before you send your pull request since they are distracting and add noise. Keep your pull requests to 100% semantic changes so that reviewers can review them more easily and move refactoring and other changes that don't modify the meaning code to separate pull requests.